addons/mass_mailing/models/mailing_list.py

```diff
@@ -223,6 +225,24 @@ def action_merge(self, src_lists, archive):
     def close_dialog(self):
         return {'type': 'ir.actions.act_window_close'}
 
+    # ------------------------------------------------------
+    # MAILING
+    # ------------------------------------------------------
+
+    def _mailing_get_default_domain(self, mailing):
+        return [('list_ids', 'in', mailing.contact_list_ids.ids)]
+
+    def _mailing_get_opt_out_list(self, mailing):
```

Comment:
Method name does not really indicate that it returns record ids.
Maybe `_mailing_get_opt_out_res_ids`?

EDIT: I realized it returns mixed types, sometimes ids, sometimes emails, sometimes phone numbers?